### ISSUE TITLE
New rule proposal `sort-scripts-content`

### ISSUE THREAD

**Issue by axel7083** [sveltejs/eslint-plugin-svelte#1426]
### Motivation

When dealing in large codebase, readability is very important and svelte component may be complex depending on the number of states, function etc.

### Description

To address this issue, similar to eslint rule sorting imports I suggest a rule that would automatically sort the content based on user orders (with some nice default)

### Examples

```svelte
<!-- ✗ BAD -->
<script lang="ts">
	let count: Foo = $state({ bar: 'dummy' });
	interface Foo {
		bar: string;
	}
</script>

<!-- ✓ GOOD -->
<script>
	interface Foo {
		bar: string;
	}
	let count: Foo = $state({ bar: 'dummy' });
</script>
```

```svelte
<!-- ✗ BAD -->
<script lang="ts">
	type Foo = string;
	import Bar from './bar.svelte';

	function hello(): string {
		return 'world';
	}
</script>

<!-- ✓ GOOD -->
<script>
	import Bar from './bar.svelte';
	type Foo = string;

	function hello(): string {
		return 'world';
	}
</script>
```

### Additional comments

To experiment with Eslint rules and have some fun I implemented it, I created a draft PR https://github.com/sveltejs/eslint-plugin-svelte/pull/1427

if this proposal get accepted I can refine & improve the code I wrote, otherwise closing it !

**Comment by marekdedic** [comment]
Hi, I've been wanting to experiment with a similar rule for some time, thanks! I glanced over the PR and I think it should be a little more complicated. Non-exhaustive list of categories I'd like to be able to order:
- state variables
- derived variables
- interface declarations
- Props interface declaration
- props passing
- @tanstack/svelte-query query definitions

Moreover, sometimes I'd need to break the order, e.g. when things depend on each other, they need to be in certain order...

**Comment by sacrosanctic** [comment]
This is probably a no go. Simple components can be sorted manually. Complex apps have many interdependencies, which means you'll need to mark them as exceptions.

**Comment by axel7083** [comment]
> Complex apps have many interdependencies, which means you'll need to mark them as exceptions.

@sacrosanctic could you elaborate and provide some examples? I am not sure to see what kind of _interdependencies_ you talk about that would prevent this?

To give more details, this only sort the JS inside the `<scripts>`, you cannot have more than one `<scripts>` per .svelte file, and the JS do not need to be ordered before being reference, so you can have a state using a function, and have the function bellow.

For us, this rule would help on the readability of components, where we often end up with states declared at the end, of types/interfaces just above the function that needed it; this rule would ensure that users can specify the order they want
